app/models/attendance.rb

```diff
@@ -5,7 +5,7 @@ class Attendance < UuidApplicationRecord
   before_validation :round_check_in, :round_check_out
   before_validation :calc_time_in_care, if: :child_approval
   before_validation :find_or_create_service_day, if: :check_in
-  before_save :remove_absences
+  before_create :remove_absences
```

Comment:
Let's think about this use case - the likelihood of this might be 0, so feel free to correct me here.
a) is it ever possible in this code to have a service day that has TWO attendances that are both absences?
  - if not, then this is fine, I think
  - if so, we would still want to remove the *other* absence if one of the attendances get edited to be a real attendance vs. an absence, and this trigger wouldn't fire because of the edit

If it's the second case, I think what we want to do instead of changing the hook is change the behavior to remove all absences OTHER than itself when an attendance gets edited from an absence to a real attendance.

Let me know what you think about this, like I said, it's possible we have other logic preventing multiple absences from being added to the same service day, but I don't remember off the top of my head.

Reply:
That was a quick temp fix for the gif problem but I will be changing the behavior. There is nothing in the code that prevents a service day of having two attendance absences.  I was also planing on writing a hook to remove not only the other attendances with absences but all other attendances of the service day to satisfy: 

> If I change an attendance to an absence, all other attendances for that ServiceDay are deleted - we can only have one absence per day, and no other types of attendance if there's an absence

I would then change that remove absences hook to a custom validation that prevents two absences for a service day.

This case will require refactoring as well:

> if I change the check_in time to a time that belongs to a different service_day (i.e. before midnight to the day before or after midnight to the day after), the ServiceDay for the new date is created if it doesn't exist already, or the attendance is associated to the existing ServiceDay for that date
>  All absence requirements above apply here as well; if I change the check_in time to another date, and it's an absence, the above actions happen as well

Reply:
@twinsfan421 one adjustment based on the refactors I just did to unblock the deployment to prod:
```
 If I change an attendance to an absence, its time_in_care is recalculated based on the schedule for that day`
```
This is no longer an AC - instead I put the logic of determining what schedule to apply to a service day *in* the service day; attendances will either calculate their check_out - check_in, or return 0.seconds.  This feels like more fidelity to me on the part of the attendance knowing about itself.  Let me know what you think.

As for your above comments, I'm okay with putting that work in another ticket so we can get the edit endpoint out.  Do you want to write it or do you want me to?

Reply:
Ok sounds good, Ya I definitely think returning check_out - check_in, or return 0.seconds for time_in_care is the right call on that.  If you have time to write it, that would be great.  Otherwise I can do it when I get back.  I am I about to head to the airport and fly home for the holidays.  I will be available after the new year to pick back up on it.  Thanks!

Reply:
https://github.com/pieforproviders/pieforproviders/issues/2080 Made the ticket here, just used your comment since it was clearer than rewriting.  :)  Thanks!